task/sql-tasks.js

```diff
@@ -280,7 +454,20 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            CompanyName,
+            ProductName,
+            od1.UnitPrice AS "PricePerItem"
+        FROM Customers c1 JOIN Orders o1 ON o1.CustomerID = c1.CustomerID JOIN OrderDetails od1 ON od1.OrderID = o1.OrderID JOIN Products p ON p.ProductID = od1.ProductID
+        WHERE od1.UnitPrice = (
+            SELECT 
+                MAX(od2.UnitPrice) 
+            FROM Customers c2 JOIN Orders o2 ON o2.CustomerID = c2.CustomerID JOIN OrderDetails od2 ON od2.OrderID = o2.OrderID
+            WHERE c2.CompanyName = c1.CompanyName)
```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.

Reply:
also try to grup by customer id and join subquery on price and customer id